### PR TITLE
convert date to datetime

### PR DESCRIPTION
**Pull request by JulesGrd**
converting date to datetime for finding lesson with client.lessons(), more precision

### CODE REVIEW

File: pronotepy/clients.py [hunk 1]
```diff
@@ -148,7 +148,9 @@ def _login(self) -> bool:
             log.info('login failed')
             return False
 
-    def get_week(self, date: datetime.date):
+    def get_week(self, date: Union[datetime.date, datetime.datetime]):
+        if isinstance(date, datetime.datetime):
+            return 1 + int((date.date() - self.start_day).days / 7)
```
Comment (bain3) [inline pronotepy/clients.py:153]: Maybe leave it as it was. This is quite nit picky but an `isinstance` check is quite expensive. This method is only used internally and mostly with `datetime.date`, the caller can easily convert it from `datetime.datetime` outside.

File: pronotepy/clients.py [hunk 2]
```diff
@@ -302,7 +315,7 @@ def lessons(self, date_from: datetime.date, date_to: datetime.date = None) -> Li
                 output.append(dataClasses.Lesson(self, lesson))
 
         # since we only have week precision, we need to make it more precise on our own
-        return [lesson for lesson in output if date_from <= lesson.start.date() <= date_to]
+        return [lesson for lesson in output if date_from <= lesson.start <= date_to]
```
Comment (bain3) [inline pronotepy/clients.py:318]: The list ends up empty in unit tests. Not sure why but something is broken.

**Review by bain3 — CHANGES_REQUESTED** [review]
Hi, looks like you have an error in your code. I'm not sure exactly what's the issue but the method returns an empty list when it should have *at least one* element (the unit test is requesting lessons from the first 4 days).

File: pronotepy/clients.py (hunk 2 above)
Comment (JulesGrd) [reply pronotepy/clients.py:318]: done

File: pronotepy/clients.py (hunk 1 above)
Comment (JulesGrd) [reply pronotepy/clients.py:153]: the "isinstance" function does not cost that much. I join you the profiling of different calls

Comment (JulesGrd) [reply pronotepy/clients.py:153]:
```
import cProfile

cProfile.run('client.lessons(datetime.datetime.now())')
```
[restats.txt](https://github.com/bain3/pronotepy/files/7117694/restats.txt)

```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     1136    0.000    0.000    0.000    0.000 {built-in method builtins.isinstance}
```

```
import cProfile

cProfile.run('client.lessons(datetime.datetime.now(), datetime.datetime.now() + datetime.timedelta(weeks=52))')
```
[stat_52_weeks.txt](https://github.com/bain3/pronotepy/files/7117700/stat_52_weeks.txt)

```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
    46200    0.005    0.000    0.007    0.000 {built-in method builtins.isinstance}
```

**Review by bain3 — APPROVED** [review]
LGTM

**Pull request merged**